specs/sharding/beacon-chain.md

```diff
@@ -497,67 +522,74 @@ def process_operations(state: BeaconState, body: BeaconBlockBody) -> None:
 
 ```python
 def process_attestation(state: BeaconState, attestation: Attestation) -> None:
     phase0.process_attestation(state, attestation)
-    update_pending_votes(state, attestation)
+    update_pending_shard_work(state, attestation)
 ```
 
-#### `update_pending_votes`
-
 ```python
-def update_pending_votes(state: BeaconState, attestation: Attestation) -> None:
-    # Find and update the PendingShardHeader object, invalid block if pending header not in state
-    if compute_epoch_at_slot(attestation.data.slot) == get_current_epoch(state):
-        pending_headers = state.current_epoch_pending_shard_headers
-    else:
-        pending_headers = state.previous_epoch_pending_shard_headers
-    
+def update_pending_shard_work(state: BeaconState, attestation: Attestation) -> None:
     attestation_shard = compute_shard_from_committee_index(
         state,
         attestation.data.slot,
         attestation.data.index,
     )
-    pending_header = None
-    for header in pending_headers:
-        if (
-            header.root == attestation.data.shard_header_root
-            and header.slot == attestation.data.slot
-            and header.shard == attestation_shard
-        ):
-            pending_header = header
-    assert pending_header is not None
-
-    for i in range(len(pending_header.votes)):
-        pending_header.votes[i] = pending_header.votes[i] or attestation.aggregation_bits[i]
-
-    # Check if the PendingShardHeader is eligible for expedited confirmation
-    # Requirement 1: nothing else confirmed
-    all_candidates = [
-        c for c in pending_headers if
-        (c.slot, c.shard) == (pending_header.slot, pending_header.shard)
-    ]
-    if True in [c.confirmed for c in all_candidates]:
+    buffer_index = attestation.data.slot % SHARD_STATE_MEMORY_SLOTS
+    committee_work = state.shard_buffer[buffer_index][attestation_shard]
+
+    # Skip attestation vote accounting if the header is not pending
+    if committee_work.status.selector != PENDING_SHARD_DATA:
+        # TODO In Altair: set participation bit flag, if attestation matches winning header.
         return
 
-    # Requirement 2: >= 2/3 of balance attesting
-    participants = get_attesting_indices(state, attestation.data, pending_header.votes)
-    participants_balance = get_total_balance(state, participants)
+    current_headers: Sequence[PendingShardHeader] = committee_work.status.value
+
+    # Find the corresponding header, abort if it cannot be found
+    header_index = [header.root for header in current_headers].index(attestation.data.shard_header_root)
+
+    pending_header: PendingShardHeader = current_headers[header_index]
     full_committee = get_beacon_committee(state, attestation.data.slot, attestation.data.index)
-    full_committee_balance = get_total_balance(state, set(full_committee))
-    if participants_balance * 3 >= full_committee_balance * 2:
-        pending_header.confirmed = True
+    full_committee_balance = Gwei(0)
```

Comment:
(nit) I would pull this `full_committee_balance ` declaration down: right before the `for` loop to make the code clear